generated/serialization.h

```diff
@@ -0,0 +1,80 @@
+#ifndef DTOGEN_SERIALIZATION_H
+#define DTOGEN_SERIALIZATION_H
+
+#include <memory>
+#include <optional>
+
+#include "nlohmann/json.hpp"
+
+// TODO: move elsewhere
+#ifdef __clang__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif __GNUC__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif _MSC_VER
+#define UNREACHABLE_CODE __assume(0);
+#endif
+
+// TODO: move elsewhere
+#include <stdexcept>
+namespace org {
+namespace openapitools {
+namespace model {
+class invalid_enum_value : public std::runtime_error {
+    public:
+        using std::runtime_error::runtime_error;
+};
+}  // namespace org
+}  // namespace openapitools
+}  // namespace model
+
+
+namespace nlohmann {
+
+template <typename T>
+struct adl_serializer<std::optional<T>> {
+    static void to_json(json& j, const std::optional<T>& opt) {
+        opt == std::nullopt ? j = nullptr : j = *opt;
+    }
+
+    static void from_json(const json& j, std::optional<T>& opt) {
+        j.is_null() ? opt = std::nullopt : opt = j.get<T>();
+    }
+};
```

Comment:
ah... so nlohmann didn't make this automagic? really? ... again, asking for a friend.

Reply:
Yeah, I was a bit surprised that this is necessary as well. In their documentation I only found that they support STL container types out of the box. And the example for how to support custom types used `boost::optional`. So yes, I think we need this.